src/devtools/WebAudioEventObserver.ts

```diff
@@ -33,19 +53,297 @@ export class WebAudioEventObserver extends Observer<Audion.WebAudioEvent> {
       chrome.debugger.onDetach.addListener(onDetach);
       chrome.debugger.onEvent.addListener(onEvent);
 
+      this.attachInterest.increment();
+      this.eventInterest.increment();
+
       return () => {
         chrome.debugger.onDetach.removeListener(onDetach);
         chrome.debugger.onEvent.removeListener(onEvent);
-        chrome.debugger.sendCommand({tabId}, Methods.disable);
-        chrome.debugger.detach({tabId}, () => {});
+
+        this.eventInterest.decrement();
+        this.attachInterest.decrement();
       };
     });
+
+    const permission = (this.permission = new PermissionSubject());
+
+    const attachInterest = (this.attachInterest = new CounterSubject(0));
+
+    const attachSubject = new AttachStateSubject();
+
+    const eventInterest = (this.eventInterest = new CounterSubject(0));
+
+    const eventSubject = new EventStateSubject();
+
+    const debuggerState = merge(
```

Comment:
Can we have brief description of what's happening below? (1~2 sentences per operation)